casper/src/main/scala/io/casperlabs/casper/finality/MultiParentFinalizer.scala

```diff
@@ -15,13 +14,9 @@ import simulacrum.typeclass
 import io.casperlabs.storage.dag.FinalityStorage
 
 @typeclass trait MultiParentFinalizer[F[_]] {
+  def addMessage(message: Message): F[Unit]
 
-  /** Returns set of finalized blocks.
-    *
-    * NOTE: This mimicks [[io.casperlabs.casper.finality.votingmatrix.FinalityDetectorVotingMatrix]] API because we are working with the multi-parent
-    * fork choice.
-    */
-  def onNewMessageAdded(message: Message): F[Seq[FinalizedBlocks]]
+  def checkFinality: F[Seq[FinalizedBlocks]]
```

Comment:
```suggestion
  def checkFinality(): F[Seq[FinalizedBlocks]]
```
Because it can change the state.

Reply:
Done.